### ISSUE TITLE
Histories created via the REST API don't appear in the Web UI Straight Away

### ISSUE THREAD

**Issue by greenwoodma** [galaxyproject/galaxy#4605]
This is a rather odd issue, but it would appear that if you create a new history via the REST API it doesn't appear in the multiple histories view of the web UI until you create another new history; either via the REST API or using the "Create New" button in the web UI. Steps to see this in action are....

1. Firstly go and look at the current list of histories; for example on a local install visit http://localhost:8080/history/view_multiple and take note of the existing histories.
2. Create a new history via the REST API; there are many ways you could do this but possibly the easiest is from a browser with the Poster extension (in firefox or Chrome) by posting `{"name":"Testing 1, 2, 3", "deleted":false}` to http://localhost:8080/api/histories
3. Now refresh the histories page http://localhost:8080/history/view_multiple and you'll see that the history you created isn't displayed.
4. Create another new history (with a different name for sanity purposes) either via the REST API or the "Create New" button in the web UI and then refresh the page. At this point the history you created in step 2 will now appear.

I honestly have no idea why the histories don't appear until another one is created, but it's really quite frustrating when trying to debug REST API usage if the history simply doesn't appear in the web UI.

**Comment by nsoranzo** [comment]
Duplicate of https://github.com/galaxyproject/galaxy/issues/3519 ?

**Comment by dannon** [comment]
@nsoranzo I thought that at first glance, but my understanding of #3519 was that these may not be new histories even, one thing which has made that one much more problematic to track down -- the seeming randomness of it all.  This in particular should be easy to track down and fix (and given the steps above, I'm already testing it now).

**Comment by greenwoodma** [comment]
I had seen #3519 but given the case here with it being limited to those produced via the REST API I assumed it might be unrelated, and given it's easy to reliably reproduce I'm hopefully that will help someone with more knowledge of the internals to track down what's going on.

**Comment by dannon** [comment]
Yeah, I reproduced the exact error and have a few ideas.  This may well be related to the other issue, if a different presentation.  Thanks a ton for the detailed steps to cause it!

**Comment by dannon** [comment]
Ok, this is all because we make an incorrect assumption that the 'current' history is the most recently updated one, and use offsets when querying the full set of histories.  I'll work on a fix, but I bet this is going to resolve #3519 as well.

**Comment by dannon** [comment]
Should be fixed in #4610.  The assumption actually looks to have been incidental, and it was actually all due to an overcomplicated expression in which we overwrote a `0` value since it wasn't truthy.

**Comment by greenwoodma** [comment]
I've just tested the changes locally and that definitely fixes the issue I was seeing, so once the PR is merged I'm happy to say this issue can be closed.
